packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -341,6 +346,7 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
   static const bool _defaultEnableFeedback = true;
   static const TextAlign _defaultTextAlign = TextAlign.start;
 
+  TooltipController? _internalTooltipController;
```

Comment:
If I'm understanding it correctly, a controller is for the developer to control a single tooltip and is never used internally by the `Tooltip` widget. If the developer never specified a controller then why do we need to create an internal one?

Reply:
I was using MenuAnchor approach to implement the controller for the tooltip. There may be further needs for this internal controller but you are totally right.